### PR TITLE
Quickinstall macOS specific and general improvements

### PR DESCRIPTION
**Pull request by ghost**
Part of #17. General improvements proposed by shellcheck and macOS specific ones - substitution of readlink which is not portable, check for Xcode and Command Line Tools being installed.

### CODE REVIEW

**Comment by tatokis** [comment]
Just to keep track of things, as I've already told @light2yellow over IRC, 
```tatokis@tatokis-PC:~/ckb-next/ckb-next-l2y$ LANG=C ./quickinstall 
./quickinstall: line 29: cd: ".: No such file or directory
```

**Comment by tatokis** [comment]
Compiles fine on my system, I think this is clear for merging.

**Comment by ghost** [comment]
We were also talking about removing the condition that makes impossible to compile in a directory whose name contains spaces/tabs/quotes/whatever needs to be escaped but after some research I must say this is a big pain in the ass because as of 2017 neither qmake nor cmake did not provide any generic function to escape everything that needs escaping in the string when generating makefile and I'm not going to write it myself. So please test current progress and if everything is alright that'll probably be it.

File: quickinstall [hunk 1]
```diff
@@ -92,11 +114,11 @@ if [[ "$OSTYPE" == "darwin"* ]]; then
     # OSX
     newtmp
     # Run macdeployqt to avoid linking problems
-    [[ $MACDEPLOYQT == "" ]] && MACDEPLOYQT=`find ~/Qt/*/clang_64/bin ~/Applications/Qt/*/clang_64/bin /Applications/Qt/*/clang_64/bin -name macdeployqt -print -quit 2>/dev/null`
-    $MACDEPLOYQT ckb.app >/dev/null 2>$TMPFILE
+    [[ $MACDEPLOYQT == "" ]] && MACDEPLOYQT=$(find ~/Qt/*/clang_64/bin ~/Applications/Qt/*/clang_64/bin /Applications/Qt/*/clang_64/bin -name macdeployqt -print -quit 2>/dev/null)
+    $MACDEPLOYQT ckb.app >/dev/null 2>"$TMPFILE"
```
Comment (frickler24) [inline quickinstall:118]: I do not know Mac OS, but should you check that $MACDEPLOYQT is != "" ?

File: quickinstall [hunk 2]
```diff
@@ -105,32 +127,32 @@ else
     # Install apps
     sudo mkdir -p $PREFIX
```
Comment (frickler24) [inline quickinstall]: Why not surrounding "$PREFIX" in this and following lines?

File: quickinstall (hunk 1 above)
Comment (fleischie) [reply quickinstall:118]: I think the intention here is: if `$MACDEPLOYQT` is empty, set it to something, else use it's contents...

**Comment by tatokis** [comment]
Yeah, if it's blank it uses the defaults, otherwise it uses whatever the variable is set to.

File: quickinstall (hunk 1 above)
Comment (ghost) [reply quickinstall:118]: I guess this was written to be able to test hardcoded paths given to variable manually and just left untouched when everything started working. macdeployqt is pretty non-obvious and uses hardcoded paths as well (looks like a mass disease on macOS). ¯\_(ツ)_/¯

File: quickinstall (hunk 2 above)
Comment (ghost) [reply quickinstall]: You're right, it wasn't finished for some reason.

**Pull request merged**